python/pyarrow/tests/test_extension_type.py

```diff
@@ -217,3 +217,130 @@ def test_ipc_unknown_type():
     batch = ipc_read_batch(buf2)
     arr = check_example_batch(batch)
     assert arr.type == ParamExtType(3)
+
+
+class PeriodType(pa.GenericExtensionType):
+
+    def __init__(self, freq):
+        # attributes need to be set first before calling
+        # super init (as that calls serialize)
+        self.freq = freq
+        pa.GenericExtensionType.__init__(self, pa.int64(), 'pandas.period')
+
+    def __arrow_ext_serialize__(self):
+        return "freq={}".format(self.freq).encode()
+
+    @classmethod
+    def __arrow_ext_deserialize__(cls, storage_type, serialized):
+        serialized = serialized.decode()
+        assert serialized.startswith("freq=")
+        freq = serialized.split('=')[1]
+        return PeriodType(freq)
+
+    def __eq__(self, other):
+        if isinstance(other, pa.BaseExtensionType):
+            return (type(self) == type(other) and
+                    self.freq == other.freq)
+        else:
+            return NotImplemented
+
+
+@pytest.fixture
+def registered_period_type():
+    # setup
+    period_type = PeriodType('D')
+    pa.lib.register_extension_type(period_type)
+    yield
```

Comment:
`try..finally` here? Or is that not necessary with pytest?

Reply:
> try..finally here? Or is that not necessary with pytest?

AFAIU, this is not needed with pytest (it's like a unittest teardown method, the part after yield should be called in the end regardless of whether the test raised an error or not)